src/CollectionBLSImplV1.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.21;
+
+import "openzeppelin-contracts-upgradeable/contracts/proxy/utils/Initializable.sol";
+import "./IVisitable.sol";
+import "./CollectionV1.sol";
+import "./bls/BLSOwnableUpgradeable.sol";
+
+
+contract CollectionBLSImplV1 is Initializable, BLSOwnableUpgradeable, CollectionV1 {
+    bytes32 public constant EIP712_CURATE = keccak256("Curate(int256 data)");
+    bytes32 public constant EIP712_ATTACH_MUSEUM = keccak256("AttachMuseum(address museum)");
+    bytes32 public constant EIP712_DETACH_MUSEUM = keccak256("DetachMuseum(address museum)");
+    bytes32 public constant EIP712_UPDATE_NAME = keccak256("UpdateName(string name)");
+    bytes32 public constant EIP712_UPDATE_DECIMALS = keccak256("UpdateDecimals(uint8 decimals)");
+    bytes32 public constant EIP712_UPDATE_VERSION = keccak256("UpdateVersion(uint256 version)");
+
+    function initialize(address _museum, uint256[4] memory blsPublicKey_, string memory _name, uint8 _decimals, uint256 _version) initializer public {
```

Comment:
Being picky but I think we should put the blsPublicKey_ at the end of the args, keeping the basic collection params together. Also why did we name it with the _ at the end? Was that on purpose?

Reply:
ah actually i was just following the openzeppelin convention due to habit and copied the rest xD. I will change it to _blsPublicKey to be consistent with our convention